tests/roundtrip.rs

```diff
@@ -0,0 +1,144 @@
+extern crate tempfile;
+extern crate escargot;
+extern crate reqwest;
+extern crate unzip;
+
+use tempfile::tempdir;
+use std::fs::File;
+//use std::path::Path;
+use std::io::{self, Write};
+use std::vec::Vec;
+//use std::env;
+
+#[macro_use]
+extern crate duct;
+
+//use duct::cmd;
+
+#[test]
+fn roundtrip() { //return -> std::io::Result<()> ?
+//TODO: wrap test tasks in a function that returns Result so we can safely
+//dispose of temp directories, and kill the server if test fails?
+
+    //let mut file_vec = Vec::new();
+
+
+  //generate some temporary files in a directory
+  let dir= tempdir().unwrap();
+  let archive_dir = tempdir().unwrap();
+    {
+        let dir_path = dir.path();//.to_owned();
+        let archive_dir_path = archive_dir.path().to_str().unwrap();
+
+        //let file = tempfile::tempfile_in(dir_path).unwrap();
+
+        for i in 1..100 {
+            // let mut file = tempfile::tempfile_in(dir_path).unwrap();
+
+            //file_vec.push(file);
+            let file_path= format!("{}/file{}", dir_path.to_str().unwrap(), i);
+            let mut file = File::create(file_path).unwrap();
+            writeln!(file, "here's some file contents");
+
+        }
+
+        //build and run this file with escargot:
+        //assumption: defaults to building in ./target/debug
+        escargot::CargoBuild::new()
+            .current_target()
+            .exec()
+            .unwrap();
+
+        let mut archive_path =archive_dir.path().to_str().unwrap().to_owned();
+        archive_path.push_str("/docs.archive");
+
+        //use duct to run the newly built binary,
+        //and create docs.archive in directory for that binary
+        let run_status = cmd!("./target/debug/static-filez", "build",
+             dir.path().to_owned(), &archive_path )
+             .read();//.unwrap();
+        match run_status {
+            Ok(o)=> {println!("{}", o)}
+            Err(e)=> {
+                panic!("Failed to archive files in {:?}: Error: {}",
+                dir.path().to_owned().to_str(), e)
+            }
+        }
+
+
+        //NOTE: must use .start() or test will hang indefinitely
+        // since Result is only generated after server process
+        // terminates for .run()
+        let process_handle;
+        let server_status = cmd!("./target/debug/static-filez",
+             "serve", "-p", "3000", &archive_path)
+             .start();
```

Comment:
Might be a good idea to use a random port/socket here as tests may run in parallel. The `-p` flag is actually from https://docs.rs/clap-port-flag/0.3.0/clap_port_flag/struct.Port.html so there might be a way to supply this more programmatically (but I've never done that).